api_tests/waffle/views/test_waffle_cookies.py

```diff
@@ -0,0 +1,83 @@
+import mock
+import pytest
+import waffle
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    FlagFactory,
+    ProjectFactory
+)
+from website.ember_osf_web.decorators import camouflage_request
+
+from website.project.views.node import _view_project
+from functools import wraps
+from api.base.views import root
+from framework.auth.core import _get_current_user
+from tests.json_api_test_app import JSONAPITestApp
+from tests.base import OsfTestCase
+
+
+class MockUser(object):
+    is_authenticated = False
+
+
+def check_for_test_flag(func):
+    @wraps(func)
+    def wrapped(self, *args, **kwargs):
+        from flask import request
+        request.user = _get_current_user() or MockUser()
+        request = camouflage_request(request)
+        print waffle.flag_is_active(request, 'test_flag')
+        return func(self, *args, **kwargs)
+    return wrapped
+
+
+@pytest.mark.django_db
+class TestWaffleCookies(OsfTestCase):
+
+    def setUp(self):
+        super(TestWaffleCookies, self).setUp()
+        self.flag = FlagFactory(name='test_flag')
+        self.flag.percent = 50
+        self.flag.everyone = None
+        self.flag.save()
+
+    @pytest.mark.enable_bookmark_creation
+    @mock.patch('website.project.views.node._view_project', check_for_test_flag(_view_project))
```

Comment:
check_for_test_flag whole purpose is actually [this one line](https://github.com/CenterForOpenScience/osf.io/pull/9084/files#diff-6e82686af75c2fb55d824be41c7837deR30) so I'm able to basically insert the line `waffle.flag_is_active(request, 'test_flag')` before other code executes in the patched view, so can check `test_flag` without actually testing a flag in the wild.

Reply:
Ok, that makes sense as a kind of roundabout unit test for the `OsfWebRenderer` code that actually sets the cookie. 👍 

Ideally, we'd also catch anytime someone calls `waffle.flag_is_active` with a bare flask request (like those four I found manually). Not sure the best way to approach that... maybe have an auto-used test fixture that mocks/wraps `waffle.flag_is_active` to assert that `request.user` and `request.COOKIES` exist?

Reply:
I don't want to modify waffle S.O.P. more than we have already. I think it'd be better to just remove those flags, many of which we not longer use/need. I don't want to write code/tests to work with routes we will just have to delete eventually. But I understand the appeal of having `flag_is_active` working in all contexts with the same function. So I'm persuadable.

Reply:
Removing flags would be good too! I think the value of wrapping `flag_is_active` would be to catch any mistakes in the future, since dynamic types make it so easy to pass a flask request without realizing it's a problem, and the error might only surface much later (when someone tries to set the flag to something other than "everyone on/off" and starts getting 500s).

Buuut if that's not easy, not a big deal.